### ISSUE TITLE
RuntimeError: Cannot open file

### ISSUE THREAD

**Issue by yuanjie-ai** [docarray/docarray#1407]
How to avoid errors caused by repeated execution

```python
from docarray import DocList, BaseDoc
from docarray.index import HnswDocumentIndex, ElasticV7DocIndex, ElasticDocIndex
import numpy as np
from docarray.typing import ImageUrl, ImageTensor, NdArray


class ImageDoc(BaseDoc):
    url: ImageUrl
    tensor: ImageTensor
    embedding: NdArray[128]


# create some data
dl = DocList[ImageDoc](
    [
        ImageDoc(
            url="https://upload.wikimedia.org/wikipedia/commons/2/2f/Alpamayo.jpg",
            tensor=np.zeros((3, 224, 224)),
            embedding=np.random.random((128,)),
        )
        for _ in range(100)
    ]
)

# create a Document Index
index = HnswDocumentIndex[ImageDoc](work_dir='test_index')


# index your data
index.index(dl)

# find similar Documents
query = dl[0]
results, scores = index.find(query, limit=10, search_field='embedding')
```

**Comment by JoanFM** [comment]
what is the errlr you see?

**Comment by samsja** [comment]
cc @JohannesMessner we talk about it earlier.

Basically it is not a"bug" in a sense that is expected to fail is the file already exist. 

On the other hand it is true that it is a pain to work with because we need to always clean the folder.

Maybe we could override the files each time? But this could lead to a lot of problems where we erase important data by mistakes.

**Comment by JoanFM** [comment]
The potential `bug` I see is the fact that u cannot reload and keep adding Documents there, we cannot start from an already present Folder. This could be a bug actually

**Comment by JoanFM** [comment]
What us your intended behavior @yuanjie-ai? to keep incrementally indexing documents?

**Comment by JoanFM** [comment]
Closed by #1277
